backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/VersionController.kt

```diff
@@ -1,20 +1,50 @@
 package com.tngtech.apicenter.backend.connector.rest.controller
 
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper
+
+import com.tngtech.apicenter.backend.domain.entity.Version
 import com.tngtech.apicenter.backend.connector.rest.dto.VersionDto
 import com.tngtech.apicenter.backend.connector.rest.mapper.VersionDtoMapper
 import com.tngtech.apicenter.backend.domain.handler.VersionHandler
 import org.springframework.web.bind.annotation.DeleteMapping
 import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestMethod
+import org.springframework.web.bind.annotation.RequestHeader
 import org.springframework.web.bind.annotation.RestController
+import org.springframework.http.MediaType
 import java.util.UUID
+import javax.servlet.http.HttpServletResponse
+import mu.KotlinLogging
+
+private val logger = KotlinLogging.logger {}
 
 @RestController
 class VersionController constructor(private val versionHandler: VersionHandler, private val versionDtoMapper: VersionDtoMapper) {
 
-    @GetMapping("/specifications/{specificationId}/versions/{version}")
-    fun findVersion(@PathVariable specificationId: UUID, @PathVariable version: String): VersionDto = versionDtoMapper.fromDomain(versionHandler.findOne(specificationId, version))
+    @RequestMapping("/specifications/{specificationId}/versions/{version}",
+        produces = ["application/json", "application/yml"],
+        headers = ["Accept=application/json", "Accept=application/yml"],
+        method = [RequestMethod.GET])
+    fun findVersion(@PathVariable specificationId: UUID,
+                    @PathVariable version: String,
+                    @RequestHeader(value="Accept", defaultValue=MediaType.APPLICATION_JSON_VALUE) accept: String = MediaType.APPLICATION_JSON_VALUE): VersionDto {
+        // i.e. The integration test and unit test require the default specified in two different ways
+        val foundVersion = versionHandler.findOne(specificationId, version)
+        if (accept == "application/yml") {
```

Comment:
Instead of doing the content type negotiation by hand, I'd suggest using something like this: https://blog.jdriven.com/2016/11/handling-yaml-format-rest-spring-boot/ Let's add a new `ObjectMapper` for yaml and configure Spring Web MVC to use it as a possible `MessageConverter`. Then you don't have to do this in every method and it works as expected.